fvm/src/kernel/mod.rs

```diff
@@ -245,7 +246,7 @@ pub trait CryptoOps {
     /// `digest_out`, returning the size of the digest written to `digest_out`. If `digest_out` is
     /// to small to fit the entire digest, it will be truncated. If too large, the leftover space
     /// will not be overwritten.
-    fn hash(&mut self, code: u64, data: &[u8]) -> Result<[u8; 32]>;
+    fn hash(&mut self, code: u64, data: &[u8]) -> Result<MultihashGeneric<32>>;
```

Comment:
Note: this should not return a multihash, but rather the digest itself, i.e. no self-describing prefix. This should probably be a Vec<u8> to account for future digest functions returning more than 32 bytes.

Reply:
This can be bumped to 64 bytes as well, is there an expectation that we will be adding larger hashes than that in the future that we would need to allocate for? We don't care for the prefix, true, but we do care about owned bytes and a length (can also pass in mutable bytes). Using an existing struct for that is easy and we dont have to resort to allocating.

Reply:
I would bump to 64 bytes (512 hashes aren't uncommon) and leave it at that.

Reply:
Makes sense.